app/models/ability.rb

```diff
@@ -1,47 +1,32 @@
-# frozen_string_literal: true
-
 class Ability
   include CanCan::Ability
 
   def initialize user
-    user ||= User.new # guest user (not logged in)
-
-    if user.role_admin?
-      can :manage, :all
-    else
-      can :read, Room
-      can :read, Amenity
-      can :read, Review
-      can :read, RoomType
-      can :read, RoomAvailability
-      if user.present?
-        user_role_user
-        user_role_booking
-        user_role_request_review
-      end
-    end
+    user ||= User.new
+
+    return if user.blank?
+
+    define_user_permissions user
+
+    return unless user.role_admin?
+
+    can :manage, :all
+    can :access, :admin
```

Comment:
```suggestion
    return if user.blank?
    
    if user.role_user?
      define_user_permissions user
    elsif user.role_admin?
      can :manage, :all
      can :access, :admin
    end
```
E chia ra thành if elsif theo role, để tách quyền cho từng role. Tránh bị lặp code nhé